tests/unit_tests/blas/batch/gemm_batch_stride.cpp

```diff
@@ -48,12 +48,65 @@ extern std::vector<sycl::device *> devices;
 namespace {
 
 template <typename fp>
+typename std::enable_if<std::is_integral<fp>::value, bool>::type check_equal_int(fp x, fp x_ref,
+                                                                                 int error_mag) {
+    return (std::abs(x - x_ref) <= 1);
+}
+
+template <class T>
+struct acc_type {
+    typedef host_accessor<T, 1, access_mode::read> type;
+};
+
+template <class T>
+using acc_type_t = typename acc_type<T>::type;
+
+template <class T>
+struct vec_type {
+    typedef vector<T, allocator_helper<T, 64>> type;
+};
+
+template <class T>
+using vec_type_t = typename vec_type<T>::type;
+
+// Check for int32_t and Ts=float as small differences in the reference become large after rounding
+bool check_equal_matrix_int(acc_type_t<int32_t> &M, vec_type_t<int32_t> &M_ref,
```

Comment:
This seems to be duplicated on 3 files. Is it possible to move `check_equal_matrix_int`, `check_mat` and the helper types in `test_common.hpp`?
I think just `check_equal_matrix_int` will need to be `inline` to make sure it is not duplicated in multiple translation units.

Reply:
I have moved this into `test_common.hpp` under `check_almost_equal_matrix`